### PR TITLE
Shrink completion logic

### PR DESCRIPTION
**Pull request by jdfreder**

### CODE REVIEW

File: src/notebook/components/cell/editor.js [hunk 1]
```diff
@@ -72,10 +70,26 @@ export default class Editor extends React.Component {
...
+      },
+    }));
+  }
+
   render() {
-    const extraKeys = {"Ctrl-Space": "autocomplete"}
-    let cmp = (...args) => {return completions(this._complete, ...args);}
-    cmp.async = true;
+    const extraKeys = { 'Ctrl-Space': 'autocomplete' };
+    const hint = this.completions.bind(this);
```
Comment (rgbkrk) [inline src/notebook/components/cell/editor.js]: This binding should be moved up out of `render` and into the constructor.

Comment (jdfreder) [inline src/notebook/components/cell/editor.js]: :+1:

File: src/notebook/components/cell/editor.js [hunk 2]
```diff
@@ -72,10 +70,26 @@ export default class Editor extends React.Component {
...
+    hint.async = true;
```
Comment (rgbkrk) [inline src/notebook/components/cell/editor.js]: I suppose this too.

Comment (jdfreder) [inline src/notebook/components/cell/editor.js]: :heavy_check_mark:

**Comment by jdfreder** [comment]
Ironically this PR net adds one line of code, but it's because of the dictionary expansion

**Comment by Carreau** [comment]
The separate file was on purpose to add extra logic later for multiple completer, but fair enough.

**Pull request merged**